src/Text/Pandoc/Writers/XWiki.hs

```diff
@@ -0,0 +1,256 @@
+
+inlineToXWiki LineBreak = return "\n"
+
+-- FIXME: better understand this
+inlineToXWiki SoftBreak = return " "
+
+inlineToXWiki (Emph lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "//" <> contents <> "//"
+
+inlineToXWiki (Strong lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "**" <> contents <> "**"
+
+inlineToXWiki (Strikeout lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "--" <> contents <> "--"
+
+inlineToXWiki (Superscript lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "^^" <> contents <> "^^"
+
+inlineToXWiki (Subscript lst) = do
+  contents <- inlineListToXWiki lst
+  return $ ",," <> contents <> ",,"
+
+-- TODO: Not supported. Maybe escape to HTML?
+inlineToXWiki (SmallCaps lst) = do
+  contents <- inlineListToXWiki lst
+  return contents
+
+inlineToXWiki (Quoted SingleQuote lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "'" <> contents <> "'"
+
+inlineToXWiki (Quoted DoubleQuote lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "\"" <> contents <> "\""
+
+inlineToXWiki (Code (_,classes,_) contents') = do
+  let at  = Set.fromList classes `Set.intersection` highlightingLangs
+  let contents = pack contents'
+  return $
+    case Set.toList at of
+      [] -> "{{code}}" <> contents <> "{{/code}}"
+      (l:_) -> "{{code language=\"" <> (pack l) <> "\"}}" <> contents <> "{{/code}}"
+
+inlineToXWiki (Cite _ lst) = inlineListToXWiki lst
+
+-- FIXME: optionally support this (plugin?) 
+inlineToXWiki (Math _ str) = return $ "{{formula}}" <> (pack str) <> "{{/formula}}"
+
+inlineToXWiki il@(RawInline frmt str)
+  | frmt == Format "xwiki" = return $ pack str
+  | otherwise              = "" <$ report (InlineNotRendered il)
+
+-- TODO: Handle anchors
+inlineToXWiki (Link _ txt (src, _)) = do
+  label <- inlineListToXWiki txt
+  case txt of
+     [Str s] | isURI src && escapeURI s == src -> return $ pack src
+     _  -> return $ "[[" <> label <> ">>" <> (pack src) <> "]]"
+
+inlineToXWiki (Image _ alt (source, tit)) = do
+  alt' <- inlineListToXWiki alt
+  let
+    titText = pack tit
+    params = intercalate " " [
+        if Text.null alt' then "" else "alt=\"" <> alt' <> "\"",
+          if Text.null titText then "" else "title=\"" <> titText <> "\""
+        ]
+  return $ "[[image:" <> (pack source) <> (if Text.null params then "" else "|| " <> params) <> "]]"
+
+inlineToXWiki (Note contents) = blockListToXWiki contents
+
+-- FIXME: support attrs
+inlineToXWiki (Span _ contents) = inlineListToXWiki contents
+  
+-- Utility method since (for now) all lists are handled the same way
+blockToXWikiList :: PandocMonad m => Text -> [[Block]] -> XWikiReader m Text
+blockToXWikiList marker contents = do
+  lev <- asks listLevel
+  contents' <- local (\s -> s { listLevel = listLevel s <> marker } ) $ mapM listItemToXWiki contents
+  return $ vcat contents' <> if Text.null lev then "\n" else ""
+  
+
+listItemToXWiki :: PandocMonad m => [Block] -> XWikiReader m Text
```

Comment:
If this doesn't handle loose lists (you say numbering restarts), then one solution might be to strip trailing newlines from the items, so they don't end up with blank lines in them even if paragraphs are used. This will still break when the items have multiple blocks in them, of course, but it would be a lot better in common cases.

Reply:
Thanks, that seems to work better.